libparsec/crates/account/src/lib.rs

```diff
@@ -28,6 +33,26 @@ pub struct Account {
 pub enum AccountFromPasswordError {
     #[error("Server provided an invalid password algorithm config: {0}")]
     BadPasswordAlgorithm(CryptoError),
+    #[error("Component has stopped")]
+    Stopped,
+    #[error("Cannot communicate with the server: {0}")]
+    Offline(#[from] ConnectionError),
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum AccountSendEmailValidationTokenError {
+    #[error("Email recipient refused")]
```

Comment:
When sending an email to invite a user/device, the status for the email are:
```rust
pub enum InvitationEmailSentStatus {
    Success,
    ServerUnavailable,
    RecipientRefused,
}
```

So why here do we also have `InvalidEmail` and `EmailParseError` ?
And, beside, `InvalidEmail` and `EmailParseError` seem very similar, shouldn't we merge them together ?

Reply:
The email parse error comes from the the emailAdress parser, it provides more detail on which part has failed

And actually the server never responds invalid email (because we check it first) so we can get rid of invalid email altogether